src/Jackett.Common/Indexers/Pretome.cs

```diff
@@ -188,93 +133,78 @@ public override async Task<IndexerConfigurationStatus> ApplyConfiguration(JToken
+
             // Get result from redirect
+            var loginCookies = result.Cookies;
             await FollowIfRedirect(result, LoginUrl, null, loginCookies);
 
-            await ConfigureIfOK(loginCookies, result.Content != null && result.Content.Contains("logout.php"), () =>
-            {
-                CookieHeader = string.Empty;
-                throw new ExceptionWithConfigData("Failed", configData);
-            });
+            await ConfigureIfOK(loginCookies, result.Content?.Contains("logout.php") == true,
+                                () => throw new ExceptionWithConfigData("Failed", configData));
 
             return IndexerConfigurationStatus.RequiresTesting;
         }
 
         protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query)
         {
             var releases = new List<ReleaseInfo>();
-            var queryUrl = SearchUrl;
-            var queryCollection = new NameValueCollection();
-            var cats = MapTorznabCapsToTrackers(query);
-            var tags = string.Empty;
-            var catGroups = new List<string>();
-            foreach (var cat in cats)
-            {
-                //"cat[]=7&tags=x264"
-                var cSplit = cat.Split('&');
-                if (cSplit.Length > 0)
-                {
-                    var gsplit = cSplit[0].Split('=');
-                    if (gsplit.Length > 1)
-                    {
-                        catGroups.Add(gsplit[1]);
-                    }
-                }
 
-                if (cSplit.Length > 1)
-                {
-                    var gsplit = cSplit[1].Split('=');
-                    if (gsplit.Length > 1)
-                    {
-                        if (tags != string.Empty)
-                            tags += ",";
-                        tags += gsplit[1];
-                    }
-                }
-            }
-
-            if (catGroups.Distinct().Count() == 1)
+            var qc = new List<KeyValuePair<string, string>>(); // NameValueCollection don't support cat[]=19&cat[]=6
+            if (query.IsImdbQuery)
             {
-                queryCollection.Add("cat[]", catGroups.First());
+                qc.Add("search", query.ImdbID);
+                qc.Add("st", "1");
+                qc.Add("sd", "1");
             }
-
-            if (!string.IsNullOrWhiteSpace(query.GetQueryString()))
+            else if (!string.IsNullOrWhiteSpace(query.GetQueryString()))
             {
-                queryCollection.Add("st", "1");
-                queryCollection.Add("search", query.GetQueryString());
+                qc.Add("search", query.GetQueryString());
+                qc.Add("st", "1");
             }
 
-            // Do not include too many tags as it'll mess with their servers.
-            if (tags.Split(',').Length < 7)
+            // parse categories and tags
+            var catGroups = new List<string>();
+            var tagGroups = new List<string>();
```

Comment:
If we're going to call distinct at the end anyway, might as well make this `HashSet<string>` and remove lines 185&186. HashSet and List behave the same in the remaining code.

Reply:
done